python/tvm/relay/op/contrib/forge/forge_passes.py

```diff
@@ -3953,7 +3956,183 @@ def callback(self, pre, post, node_map):
                 num_newaxis-=1
             return act
         return post
+    
+class DecomposeGridSample(DFPatternCallback):
```

Comment:
Let's add documentation to this function that explains what it does, and why. Also, put here a reference regarding how decomposition is done. E.g. link you shared in PR description.